lib/logstash/inputs/eventlog.rb

```diff
@@ -50,8 +50,12 @@ def run(queue)
 
       events = @wmi.ExecNotificationQuery(wmi_query)
 
-      while
-        notification = events.NextEvent
+      while true
+        begin
+          notification = events.NextEvent(1000) #timeout is 1000 ms
+        rescue
+          next
+        end
```

Comment:
This begin;rescue block is too broad here and not needed, because we are already rescuing all the errors at the end of this method.